### DISCUSSION TITLE
✔️Add Buy Store Elite Hero Soulstones

### DISCUSSION THREAD

**Discussion by delgatojr** [zebscripts/AFK-Daily#49] [Ideas]
I see them pop up in the store from time to time. It's like 90 gems for 5 Elite Hero Soulstones.
I think this is good value.

They pop up in the top row, right three slots in Store.

**Comment by Zebiano** [comment]
I definitely want to add this to the store! And I actually think they're always on the same tile, which means a simple RGB check on that tile to see if it's a purple stone should be enough to implement this.

**Reply by delgatojr** [reply]
I got two at the same time, once. Not sure which tiles exactly. Top row, for sure.

**Reply by Zebiano** [reply]
Let's keep an eye on it and try to get all the possible coords and RGB values.

I've refreshed the daily store a few times and got one on the top row, 4th tile: `910 850 9e62d0`

**Reply by delgatojr** [reply]
![image](https://user-images.githubusercontent.com/78246856/116006468-467f8c00-a5c0-11eb-88b3-7f16fb405616.png)
Top row, 3rd tile: `650 850 544077`

**Reply by delgatojr** [reply]
![image](https://user-images.githubusercontent.com/78246856/116006556-9c543400-a5c0-11eb-87d8-d93995a648fb.png)
Top row, 2nd tile: `410 850 9787c9`

**Reply by delgatojr** [reply]
> 
> 
> Let's keep an eye on it and try to get all the possible coords and RGB values.
> 
> I've refreshed the daily store a few times and got one on the top row, 4th tile: `910 850 9e62d0`
![image](https://user-images.githubusercontent.com/78246856/116016771-8495a380-a5f2-11eb-9cb6-a4c5c0d890f2.png)

Top row, fourth tile, I got: `910 850 a266d4`?

**Reply by delgatojr** [reply]
Now I got top row, fourth tile: `910 850 a569d7`?

**Reply by Zebiano** [reply]
What. Are you really getting different values everytime...? There has to be something wrong there, that can't happen. I doubt Lilith moves the tile around inside each position... I'll keep testing it in the future and see if I get inconsistencies as well.

**Reply by kevingrillet** [reply]
![Bluestacks_2021-04-29_21-07_hT1OL](https://user-images.githubusercontent.com/7203617/116604784-dca01300-a92e-11eb-8862-1d19768d7409.png)
`[DEBUG] getColor 910 850 > RGB: a569d7`

I think the last patch did change some values... [old](https://www.google.fr/search?q=%239e62d0) vs [new](https://www.google.com/search?q=%23a569d7)

**Reply by delgatojr** [reply]
It would change each time there's a new stone. I'd buy one and the next one would have a new RGB value.

**Reply by kevingrillet** [reply]
```shell
#!/bin/bash
# ./utils/utils_colors.sh
# https://www.shellcheck.net/

# sRGBColorDelta <COLOR1> <COLOR2>
# https://stackoverflow.com/a/22692625/7295428
# 0 means opposit colors, 100 means same colors
sRGBColorDelta() {
    if [ "$#" -ne 2 ] ; then
        echo "Usage: sRGBColorDelta <COLOR1> <COLOR2>" >&2
        echo " 0 means opposit colors, 100 means same colors" >&2
        return 1
    fi
    r=$((0x${1:0:2} - 0x${2:0:2}))
    g=$((0x${1:2:2} - 0x${2:2:2}))
    b=$((0x${1:4:2} - 0x${2:4:2}))
    d=$((((765 - (${r#-} + ${g#-} + ${b#-})) * 100) / 765))                     # 765 = 3 * 255
    echo $d
}

sRGBColorDelta 9e62d0 a266d4
```

output: `98`

**Reply by delgatojr** [reply]
That looks like some genius stuff. I like!

**Reply by kevingrillet** [reply]
Well, the risk is to detect another really close purple  square as the Soulstone 😨

**Reply by delgatojr** [reply]
Hmm. I don't think that will be a problem, because this will check the value of the coordinates and the coordinates will be the Soulstone, right?

**Reply by kevingrillet** [reply]
I need to test with Hero's Essence at the same position to see the delta... I hope it's less than 98...

**Reply by Zebiano** [reply]
It's really nice to see this function in use, especially with a concrete example! But I'd like to implement that function at the end of this recent wave of PRs to make sure it works the way it's intended. And we'll also have to test out the best delta value(s), maybe they'll even have to be specific to different parts of the script. It'll be fun nonetheless! Let's just be patient for now.

**Reply by kevingrillet** [reply]
I did add it during my refacto, it's working at the moment.

**Reply by Zebiano** [reply]
Just to keep this updated, I got the following for tile 2 and 3:
`650 850` 57447b
`910 850` a569d7
The fourth tile matches with [this one ](https://github.com/zebscripts/AFK-Daily/discussions/49#discussioncomment-657033) and [this one](https://github.com/zebscripts/AFK-Daily/discussions/49#discussioncomment-676408), so I'd say it's safe for now.
